app/services/index_deleter.rb

```diff
@@ -0,0 +1,37 @@
+class IndexDeleter
+  def initialize(args)
+    @index_alias = args[:index_alias]
+    @client = args[:client]
+    @logger = args[:logger]
+  end
+
+  def run
+    aliases = client.indices.get_aliases.keys
+    indexes_to_be_deleted = prepare_indexes_for_deletion aliases
+    delete indexes_to_be_deleted
```

Comment:
nitpicking here, but the use of parentheses are more idiomatic in Ruby.

`delete(indexes_to_be_deleted)`